pandas/_libs/tslibs/src/datetime/np_datetime.c

```diff
@@ -1091,3 +1091,55 @@ PyArray_DatetimeMetaData
 get_datetime_metadata_from_dtype(PyArray_Descr *dtype) {
     return (((PyArray_DatetimeDTypeMetaData *)dtype->c_metadata)->meta);
 }
+
+
+// we could use any hashing algorithm, this is the original CPython's for tuples
```

Comment:
We have these same defines in `khash_python.h` already. Not sure about duplicating here versus refactoring

Reply:
Yah I moved some of it rather than duplicating, likely could go further down that path.  I'd like to get this in soonish so prefer to do bigger refactors separately

Reply:
I think with C its generally really hard to track down the effects of copy/pasting defines across headers and implementations though. Maybe we just create a `python_hash.h` file that khash and np can include?

Reply:
Moving it into its own file seems nice probably.  I wouldn't care too much, since this is only a problem if you would include the other header here (even then it might not be since it matches).

On a general note, it might be good to add include guards to your headers:
```
#ifndef PD_LIBS_..._CPYTHON_HASH_H
#define PD_LIBS_..._CPYTHON_HASH_H

<header content>

#endif  /* PD_LIBS_..._CPYTHON_HASH_H */
```
(some pattern there, google style suggests full paths and at least a partial path would make sense I think)